lib/formats/sh.sh

```diff
@@ -159,37 +159,48 @@ original_check() {
     fi
 }
 
-cp_hardlink() {
+cp_symlink() {
     print_progress_prefix
-    echo "${COL_YELLOW}Hardlinking to original: ${COL_RESET}$1"
+    echo "${COL_YELLOW}Symlinking to original: ${COL_RESET}$1"
     if original_check "$1" "$2"; then
         if [ -z "$DO_DRY_RUN" ]; then
-            # If it's a directory cp will create a new copy into
-            # the destination path. This would lead to more wasted space...
-            if [ -d "$1" ]; then
-                rm -rf "$1"
-            fi
-            cp --remove-destination --archive --link "$2" "$1"
+            # replace duplicate with symlink
+            rm -rf "$1"
+            ln -s "$2" "$1"
+            # make the symlink's mtime the same as the original
+            touch -mr "$2" -h "$1"
         fi
     fi
 }
 
-cp_symlink() {
+cp_hardlink() {
+    if [ -d "$1" ]; then
```

Comment:
This might not be very obvious to the user. I think the proper way would be to replace every file in the directory with a hardlink... That's quite a bit more complex to implement though.

Reply:
After some more thinking about this (and trying to come up with a proper solution) I changed my mind about this a bit. While it's still not very obvious for the user, it's a relatively sane way to handle those edge cases.